### PR TITLE
LPD-44834

### PR DESCRIPTION
**Pull request by rijema**
https://liferay.atlassian.net/browse/LPD-44834

The second commit is a Test.

### CODE REVIEW

**Comment by rijema** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewFilterColumnLocalServiceImpl.java [hunk 1]
```diff
@@ -126,16 +129,31 @@ private void _validate(
...
+				JSONObject jsonObject = _jsonFactory.createJSONObject(json);
+
+				String value = String.valueOf(jsonObject.get(filterType));
+
+				jsonValid = Validator.isNotNull(
+					value.replaceAll("[\\[\\]\"]", ""));
+			}
+
+			if ((filterTypeNull && !jsonNull) ||
+				(!filterTypeNull && jsonNull) || !jsonValid) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewFilterColumnLocalServiceImpl.java:156]: @rijema no need to create new variables here, please simplify by checking if the JSONArray is empty:

```
if (Validator.isNull(objectViewFilterColumn.getFilterType()) &&
	Validator.isNull(objectViewFilterColumn.getJSON())) {

	continue;
}

if ((Validator.isNull(objectViewFilterColumn.getFilterType()) &&
	 Validator.isNotNull(objectViewFilterColumn.getJSON())) ||
	(Validator.isNotNull(objectViewFilterColumn.getFilterType()) &&
	 Validator.isNull(objectViewFilterColumn.getJSON()))) {

	throw new ObjectViewFilterColumnException(
		StringBundler.concat(
			"Object field name \"",
			objectViewFilterColumn.getObjectFieldName(),
			"\" needs to have the filter type and JSON specified"));
}

JSONObject jsonObject = _jsonFactory.createJSONObject(
	objectViewFilterColumn.getJSON());

if (JSONUtil.isEmpty(
		jsonObject.getJSONArray(
			objectViewFilterColumn.getFilterType()))) {

	throw new ObjectViewFilterColumnException(
		StringBundler.concat(
			"Object field name \"",
			objectViewFilterColumn.getObjectFieldName(),
			"\" needs to have the filter type and JSON specified"));
}
```

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectViewLocalServiceTest.java [hunk 2]
```diff
@@ -688,6 +688,28 @@ private void _testAddObjectViewRelationshipFilterColumn() throws Exception {
...
+		}
+		catch (ObjectViewFilterColumnException
+					objectViewFilterColumnException) {
+
+			Assert.assertEquals(
+				StringBundler.concat(
+					"Object field name \"", objectField.getName(),
+					"\" needs to have the filter type and JSON specified"),
+				objectViewFilterColumnException.getMessage());
+		}
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectViewLocalServiceTest.java:711]: @rijema please use `AssertUtils.assertFailure`:

```
String objectFieldName = objectField.getName();

AssertUtils.assertFailure(
	ObjectViewFilterColumnException.class,
	StringBundler.concat(
		"Object field name \"", objectFieldName,
		"\" needs to have the filter type and JSON specified"),
	() -> _objectViewLocalService.addObjectView(
		TestPropsValues.getUserId(),
		_objectDefinition.getObjectDefinitionId(), false,
		LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
		Collections.singletonList(
			_createObjectViewColumn("Simon", "joses", false)),
		Collections.singletonList(
			_createObjectViewFilterColumn(
				ObjectViewFilterColumnConstants.FILTER_TYPE_INCLUDES,
				"{\"includes\": []}", objectFieldName)),
		Collections.emptyList()));
```

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewFilterColumnLocalServiceImpl.java (hunk 1 above)
Comment (rijema) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewFilterColumnLocalServiceImpl.java]: Hi. This doesn't cover the scenario we're the customer is passing a value like {"includes": [""]}, only {"includes": []}. How should we proceed?

Comment (carolmariaabb) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewFilterColumnLocalServiceImpl.java]: @rijema you should create a test with this case and adapt the code

**Comment by carolmariaabb** [comment]
ci:stop

**Comment by rijema** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
@rijema please see https://github.com/liferay-bpm/liferay-portal/pull/4012

**Pull request closed without merging**